### DISCUSSION TITLE
ValueError: [TypeError("'numpy.float32' object is not iterable",), TypeError('vars() argument must have __dict__ attribute',)]

### DISCUSSION THREAD

**Discussion by yuanjie-ai** [fastapi/fastapi#8961] [Questions] [answered]
There is "'numpy.float32'   in the return value, How to deal with this

**Comment by Mause** [comment]
Return another type instead? Numpy types aren't supported by json

**Comment by Lord-V15** [comment]
You should simply change the return type.
For example, if you were returning a np.ndarray type object, you can convert it using **tolist()**
EXAMPLE : 
```
import numpy as np

arr = np.array([11, 21, 19])
print(f'NumPy Array: {arr}')
print(type(arr))

list1 = arr.tolist()
print(f'List: {list1}')
print(type(list1))
```
This will remove all the numpy data types from the list.

**Comment by ajascha** [comment, marked as answer]
@Lord-V15 already gave the correct solution but I didn't understand it the first few times reading through it so here's a longer explanation.

This will not work since the return type isn't supported:

```
import numpy as np

@app.get("/")
async def root():
    arr = np.array([11, 21, 19])
    return {"array": arr}
```

This will work as lists are supported:

```
import numpy as np

@app.get("/")
async def root():
    arr = np.array([11, 21, 19])
    return {"array": arr.tolist()}
```

Note that this error will also occur if you are not using `numpy` explicitly but another library that is based on it, such as `pandas`.

**Comment by jarvissilva937** [comment]
If you want to know how to fix it go here: [fix object is not iterable error in python](https://pythondex.com/fixed-numpy-uint32-object-is-not-iterable)
